libraries/rush-sdk/src/index.ts

```diff
@@ -84,3 +126,43 @@ for (const property in rushLibModule) {
     });
   }
 }
+
+/**
+ * Require `@microsoft/rush-lib` under the specified folder path.
+ */
+function requireRushLibUnderFolderPath(folderPath: string): RushLibModuleType {
+  const rushLibModulePath: string = Import.resolveModule({
+    modulePath: RUSH_LIB_NAME,
+    baseFolderPath: folderPath
+  });
+
+  return require(rushLibModulePath);
+}
+
+/**
+ * Find the rush.json location and return the path, or undefined if a rush.json can't be found.
+ *
+ * @privateRemarks
+ * Keep this in sync with `RushConfiguration.tryFindRushJsonLocation`.
+ */
+function tryFindRushJsonLocation(startingFolder: string): string | undefined {
+  let currentFolder: string = startingFolder;
+
+  // Look upwards at parent folders until we find a folder containing rush.json
+  for (let i: number = 0; i < 10; ++i) {
```

Comment:
`10` is a weird limit. Apparently I wrote this code way back in 4fcb4698d8065df10a3061d0d8449ca7650736e6, but I have no idea why. 😁  If you search this code base for the regex `parent.*path.dirname` there are a number of other similar loops that don't have this limit.

(It's out of scope for this PR but I'm tempted to fix it anyway.)

Reply:
Heft also has a copy+paste of this algorithm in [RushUtilities._findRushJsonFolder()](https://github.com/microsoft/rushstack/blob/a13865bef9a20dab28c044be3504c7326bfe94b1/apps/heft/src/utilities/RushUtilities.ts#L12)

Reply:
Some of them are also doing recursion instead of loops which wastes stack.  

👉 After this PR is merged, I can make a separate PR to clean it up.

Reply:
Why bother having a limit at all?

```suggestion
  while (true) {
```

Reply:
I'm agreeing with you; just pointing out that the associated problems involve 5 or 6 files in this repo and should be a separate PR